src/pudl/metadata/classes.py

```diff
@@ -1748,3 +1758,62 @@ def to_rst(self, top_dir: DirectoryPath, csv_subdir: DirectoryPath, rst_path: st
                 rendered = encoder.to_rst(
                     top_dir=top_dir, csv_subdir=csv_subdir, is_header=header)
                 f.write(rendered)
+
+
+class DatasetteMetadata(Base):
+    """
+    A collection of Data Sources and Resources for metadata export.
+
+    Used to create metadata YAML file to accompany Datasette.
+    """
+
+    data_sources: Dict[str, DataSource]
+    resource_package: Package = Package.from_resource_ids()
+    label_columns: Dict[str, str]
+
+    @classmethod
+    def from_data_source_ids(
+        cls,
+        data_source_ids: Iterable[str] = [
+            'pudl', 'ferc1', 'eia860', 'eia860m', 'eia923'],
+        label_columns: Dict[str, str] = {
+            'plants_entity_eia': 'plant_name_eia',
+            'plants_ferc1': 'plant_name_ferc1',
+            'plants_pudl': 'plant_name_pudl',
+            'utilities_entity_eia': 'utlity_name_eia',
+            'utilities_ferc1': 'utility_name_eia',
+            'utilities_pudl': 'utility_name_pudl'
+        }
+    ) -> "DatasetteMetadata":
+        """
+        Construct a dictionary of DataSources from data source names.
+
+        Create dictionary of first and last year or year-month for each source.
+
+        Args:
+            data_source_ids: ids of data sources currently included in Datasette
+            label_columns: map from resource to column name for link label
+        """
+        data_sources = {
+            ds_id: DataSource.from_id(ds_id) for ds_id in data_source_ids}
+        ds_namespaces = set(
+            [ds.field_namespace for _, ds in data_sources.items()
+             if ds.field_namespace]
+            + ['glue'])
+        resource_package = Package(name="datasette", resources=[
+            rsrc for rsrc in Package.from_resource_ids().resources
+            if rsrc.field_namespace in ds_namespaces])
```

Comment:
I originally had this as a `from_field_namespace` constructor in the `Package` class but then it opens itself up to  fail the foreign key reference validation tests so I moved it here. Since `DatasetteMetadata` just excludes the `epacems` namespace there aren't issues with failing the validation test.

Reply:
Maybe I'm not understanding what's going on here, but it seems kind of roundabout. `Package.from_resource_ids()` constructs a package, given a list of IDs. Why not construct the list of resource IDs you want to include, and then pass that in to `Package.from_resource_ids()` directly?